### DISCUSSION TITLE
Gatsby Link: Perform high-priority prefetch on touchstart

### DISCUSSION THREAD

**Discussion by sebashine-branly** [gatsbyjs/gatsby#38588] [Ideas / Feature Requests]
Hey everyone,

By browsing https://github.com/gatsbyjs/gatsby/discussions?discussions_q=is%3Aopen+prefetch, there are lots of good ideas and suggestions that I don't want to paraphrase here but are worth a read.
My understanding of `gatsby-link` (based on the awesome documentation and by reading some discussions/issues and the source code) is that it allows pre-fetching of links on two levels.

I'll over-simplify here by pretending `touch` events are for mobile whereas `mouse` events are on desktop. I will stand corrected if this simplification is too much.

|Priority|Desktop|Mobile|`___loader` call|
|-|-|-|-|
|low|Intersection Observer API|Intersection Observer API|`enqueue`|
|high|`onmouseenter`|none|`hovering`|

Additionally, Gatsby Browser API contains [`disableCorePrefetching`](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-browser/#disableCorePrefetching) which seems to disable the low prefetch, but does not affect the high prefetch.

Piggybacking on this [comment](https://github.com/gatsbyjs/gatsby/discussions/25119#discussioncomment-346491), I think that the exact same code from https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-link/src/index.js for `onMouseEnter` could be applied to `onTouchStart`. In other words, it should listen to `touchstart` events and call `___loader.hovering`.

There are two advantages:
- without `disableCorePrefetching`, the users on touch-only devices now have a high-priority prefetch happening as well
- with `disableCorePrefetching`, the users on touch-only devices have a high-priority prefetch happening at the time the user showed a strong intent of navigating to a client-side link

In short, this feature request is about offering consistency for touch-only devices, compared to users with a device handling `mouse` events.

I am looking forward to hearing your opinion.
